lib/node_modules/@stdlib/math/base/special/gamma/src/main.c

```diff
@@ -0,0 +1,253 @@
+	}
+	return s1 / s2;
+}
+
+// END: rational_pq
+
+/* End auto-generated functions. */
+
+/**
+* Evaluates the gamma function using Stirling's formula. The polynomial is valid for \\(33 \leq x \leq 172\\).
+*
+* @private
+* @param x    input value
+* @returns    function value
+*/
+static double stirlingApprox( const double x ) {
+	double w;
+	double y;
+	double v;
+
+	w = 1.0 / x;
+	w = 1.0 + ( w * poly_p( w ) );
+	y = stdlib_base_exp( x );
+
+	// Check `x` to avoid `pow()` overflow...
+	if ( x > MAX_STIRLING ) {
+		v = stdlib_base_pow( x, ( 0.5 * x ) - 0.25 );
+		y = v * ( v / y );
+	} else {
+		y = stdlib_base_pow( x, x - 0.5 ) / y;
+	}
+	return STDLIB_CONSTANT_FLOAT64_SQRT_TWO_PI * y * w;
+}
+
+/**
+* Evaluates the gamma function using a small-value approximation.
+*
+* @private
+* @param x    input value
+* @param z    scale factor
+* @returns    function value
+*/
+static double smallApprox( const double x, const double z ) {
+	return z / ( ( 1.0 + ( STDLIB_CONSTANT_FLOAT64_EULERGAMMA * x ) ) * x );
+}
+
+/**
+* Evaluates the gamma function.
+*
+* ## Method
+*
+* 1.  Arguments \\(|x| \leq 34\\) are reduced by recurrence and the function approximated by a rational function of degree \\(6/7\\) in the interval \\((2,3)\\).
+* 2.  Large negative arguments are made positive using a reflection formula.
+* 3.  Large arguments are handled by Stirling's formula.
+*
+* ## Notes
+*
+* -   Relative error:
+*
+*     | arithmetic | domain    | # trials | peak    | rms     |
+*     |:----------:|:---------:|:--------:|:-------:|:-------:|
+*     | DEC        | -34,34    | 10000    | 1.3e-16 | 2.5e-17 |
+*     | IEEE       | -170,-33  | 20000    | 2.3e-15 | 3.3e-16 |
+*     | IEEE       | -33, 33   | 20000    | 9.4e-16 | 2.2e-16 |
+*     | IEEE       | 33, 171.6 | 20000    | 2.3e-15 | 3.2e-16 |
+*
+* -   Error for arguments outside the test range will be larger owing to error amplification by the exponential function.
+*
+* @param x    input value
+* @return     function value
+*
+* @example
+* double v = stdlib_base_gamma( 4.0 );
+* // returns 6.0
+*/
+double stdlib_base_gamma( const double x ) {
+	double sign;
+	double xc;
+	int32_t i;
+	double p;
+	double q;
+	double z;
+
+	if ( ( stdlib_base_is_integer( x ) && x < 0 ) || x == STDLIB_CONSTANT_FLOAT64_NINF || stdlib_base_is_nan( x ) ) {
+		return 0.0 / 0.0; // NaN
+	}
+	if ( x == 0.0 ) {
+		if ( stdlib_base_is_negative_zero( x ) ) {
+			return STDLIB_CONSTANT_FLOAT64_NINF;
+		}
+		return STDLIB_CONSTANT_FLOAT64_PINF;
+	}
+	if ( x > 171.61447887182298 ) {
+		return STDLIB_CONSTANT_FLOAT64_PINF;
+	}
+	if ( x < -170.5674972726612 ) {
+		return 0.0;
+	}
+	q = stdlib_base_abs( x );
+	if ( q > 33.0 ) {
+		if ( x >= 0.0 ) {
+			return stirlingApprox( x );
+		}
+		p = stdlib_base_floor( q );
+		// Check whether `x` is even...
+		i = p;
```

Comment:
```suggestion
		i = stdlib_base_floor( q );
		// Check whether `x` is even...
```

Reply:
@Pranavchiku Not clear why your changes are superior to the current code. What is missing here is explicit casting. E.g.,

```c
i = (int32_t)p;
```

Casting should (almost) always be made explicit in order to reinforce for the reader what type a receiving value is.

Additionally, just as in JS, `// comments` should always be proceeded by a blank line when the preceding line is an expression. E.g.,

```c
p = stdlib_base_floor( q );

// Check whether `x` is even...
```

Reply:
Thanks for the suggestions @kgryte and @Pranavchiku. I have made the changes now.